src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -1120,6 +1121,19 @@ public void fetchParticipationsWithSubmissionsAndResultsForCourses(List<Course>
         return getAllUsersInGroup(course, course.getInstructorGroupName());
     }
 
+    /**
+     * GET /courses/:courseId/get-title : Returns the title of the course with the given id
+     *
+     * @param courseId the id of the course
+     * @return the name/title of the course
+     */
+    @GetMapping(value = "/courses/{courseId}/get-title")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    @ResponseBody
+    public StringDTO getCourseTitle(@PathVariable Long courseId) {
+        return new StringDTO(courseRepository.getCourseTitle(courseId));
```

Comment:
what happens if the return value is null? how is this handled?

Reply:
If the return value of the repository call is `null` that same value is wrapped and sent to the client, which falls back to the ID if `undefined` or `null` is returned. I'll check what `ResponseEntity<String>` does in this case.